scripts/dts/edtlib.py

```diff
@@ -1545,25 +1545,55 @@ def _merge_includes(self, raw, binding_path):
             return raw
 
         include = raw.pop("include")
-        fnames = []
+
+        # a list of (name, allowlist, blocklist) tuples
+        include_info = []
+
+        # Simple scalar string case
         if isinstance(include, str):
-            fnames.append(include)
+            include_info.append((include, None, None))
         elif isinstance(include, list):
-            if not all(isinstance(elem, str) for elem in include):
+        # List of strings
+            if all(isinstance(elem, str) for elem in include):
+                for elem in include:
+                    include_info.append((elem, None, None))
+        # List of 'name' maps
+            elif all(isinstance(elem, dict) for elem in include) and \
+                 all(isinstance(elem.get("name"), str) for elem in include):
+                for elem in include:
+                    name = elem["name"]
+                    allowlist = elem.get("property-allowlist")
+                    blocklist = elem.get("property-blocklist")
+                    if blocklist and allowlist:
+                        _err(f"include: in {binding_path} should not specify "
+                             "both 'property-allowlist:' and 'property-blocklist:'")
+                    if allowlist and not isinstance(allowlist, list):
+                        _err(f"'property-allowlist' in {binding_path} should "
+                             "be a list")
+                    if blocklist and not isinstance(blocklist, list):
+                        _err(f"'property-blocklist' in {binding_path} should "
+                             "be a list")
+                    include_info.append((name, allowlist, blocklist))
+            else:
                 _err(f"all elements in 'include:' in {binding_path} "
-                     "should be strings")
-            fnames += include
+                     "should either be strings or maps with 'name' key')")
         else:
             _err(f"'include:' in {binding_path} "
-                 "should be a string or a list of strings")
+                 "should be a string, list of strings or map with a 'name' key")
```

Comment:
Maybe I'm confused because I'm an old-fashioned Oxford comma type, but I read this as

```
either (string) or (list of strings) or (map with a name key)
```

But above it's a list of maps, not a map.

How about "should be a string, list of strings, or list of dicts with 'name' keys"?